subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -354,6 +357,11 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
 	if (idx < argc) {
 		params->psk = argv[idx];
 		params->psk_length = strlen(argv[idx]);
+		if (params->psk_length < WIFI_PSK_MIN_LEN ||
+		    params->psk_length > WIFI_PSK_MAX_LEN) {
```

Comment:
For WPA2/3 passphrase this check works, but for WPA3/SAE password there is no upper limit, so, may be we shouldn't enforce upper boundary check for WPA3.

Reply:
I've updated the check, and moved it after security type is parsed, please have a look.